### ISSUE TITLE
[input-date-picker] Improve input date picker view at different viewports

### ISSUE THREAD

**Issue by RuiminLi98** [Esri/calcite-components#14323]
**monday.com sync:** #11835210128

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Input Date Picker

### Summary

We met an accessibility issue about this component that content is cut off when user zoom in the page to 200% 

https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/66161 

### Actual Behavior

The content is cut off when the zoom percentage increases

### Expected Behavior

The font size inside the date picker should be reduced, and the content should be fully visible without being cut off.

### Reproduction Sample

https://codepen.io/geospatialem/pen/myrZYyM

### Reproduction Steps

1. Open the [sample](https://codepen.io/geospatialem/pen/myrZYyM)
2. Zoom in to 400% zoom, which is the [1.4.10 Reflow SC](https://www.w3.org/WAI/WCAG21/Understanding/reflow) standard
3. Observe that the content gets cut off when it extends beyond the page margin.

<img width="1909" height="953" alt="Image" src="https://github.com/user-attachments/assets/bb6e1fc7-aee3-4332-a2d8-26afee9413d9" />

### Reproduction Version

5.1.0

### Working W3C Example/Tutorial

https://www.w3.org/WAI/WCAG22/Understanding/resize-text.html

### Relevant Info

https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/66161

### Regression?

NA

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise
